### ISSUE TITLE
Running SmartSeq2 logic with "-U" gives IOError for missing UMI barcodes

### ISSUE THREAD

**Issue by danielseaton** [velocyto-team/velocyto.py#56]
Hi, I am trying to run velocyto on data generated using SmartSeq2, using the current dev branch.
I ran the command:

`velocyto run -l SmartSeq2 -o ./velocyto -U $BAMPATH $GTFPATH`

And I get the following error:

```
Traceback (most recent call last):
  File "/nfs/software/stegle/users/dseaton/conda-envs/velocyto_env/bin/velocyto", line 11, in <module>
    load_entry_point('velocyto', 'console_scripts', 'velocyto')()
  File "/nfs/software/stegle/users/dseaton/conda-envs/velocyto_env/lib/python3.6/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/nfs/software/stegle/users/dseaton/conda-envs/velocyto_env/lib/python3.6/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/nfs/software/stegle/users/dseaton/conda-envs/velocyto_env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/nfs/software/stegle/users/dseaton/conda-envs/velocyto_env/lib/python3.6/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/nfs/software/stegle/users/dseaton/conda-envs/velocyto_env/lib/python3.6/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/nfs/software/stegle/users/dseaton/python_packages/velocyto.py/velocyto/commands/run.py", line 114, in run
    samtools_memory=samtools_memory, dump=dump, verbose=verbose, additional_ca=additional_ca)
  File "/nfs/software/stegle/users/dseaton/python_packages/velocyto.py/velocyto/commands/_run.py", line 156, in _run
    exincounter.peek(bamfile[0])
  File "/nfs/software/stegle/users/dseaton/python_packages/velocyto.py/velocyto/counter.py", line 148, in peek
    raise IOError("The bam file does not contain cell and umi barcodes appropriatelly formatted. If you are runnin UMI-less data you should use the -U flag.")
OSError: The bam file does not contain cell and umi barcodes appropriatelly formatted. If you are runnin UMI-less data you should use the -U flag.
```

I checked my BAM and GTF files are correct and compatible. The first few lines of my BAM are like this:

```
HS34_22606:3:2212:13187:68945	419	chr1	156025	0	122M	=	156265	344	GTCTTGCTCTATCACCTAGGCTGGACTGCAGTGCTGCAATCACAGTTAACTGCAGCCTCAACCTCCAGGGCTTGAGCAATATTCCCATCTAATTTTTATCTTGTTTAAGAAGTGCAGTCTTG	BBBBB<BFFBFFF/<F/</FF<F7FFFFFB/<BFBFFF/F<FFBFB</<<</<F<F</BF/</BFFB///<F/</B///<</</FBBB/B/FF<BF//B///<<F/B//B/<//7/BF///B	MD:Z:33G65T11A10	PG:Z:MarkDuplicates	RG:Z:22606_3#76	NH:i:6	HI:i:2	NM:i:3	AS:i:208
HS34_22606:3:2212:13187:68945	339	chr1	156265	0	21S104M	=	156025	-344	CACGGACTCCAGGGGTGCCCAGCCACGAGTAGCTCCTTTTTTTTTTTTAATATTTAGTAGAGACAAAGTGTCACCATGTTGACCAGGTTGGTGGTGATCTCCTACACTCAGGCAGTTCTCTCACC	7//BB7/<7B77/////7////</B///B7//B<<///BBB/FBF/FF</////<B/F<B/FFFBFFFFFF</<<BBB</FFBBF//BF</<</FBFFFFFFB/FBFFFFBFF<BBBB/<BBB</	MD:Z:2A3T7A15T22T50	PG:Z:MarkDuplicates	RG:Z:22606_3#76	NH:i:6	HI:i:2	NM:i:5	AS:i:208
HS34_22606:3:1314:4949:83531	99	chr1	340951	255	9M42S	=	566538	225711	GGTATCAACGCAGAGTACTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTT	//BBBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF	MD:Z:9	PG:Z:MarkDuplicates	RG:Z:22606_3#76	NH:i:1	HI:i:1	NM:i:0	AS:i:119
HS34_22606:3:2310:16017:71190	137	chr1	526895	255	8S116M	*	0	0	GGCGAAAAACGAGGACAGAAGATCGAGACCATCCTGGCTAACACGGTGAAACCCCGTCTCTACTAAAAATACAAAAAATTAGCCGGGCATGGTAGTGGGCGCCTGTAGTCCCAGCTACTTGGGA	/////<//<//<////<//B/F/</BB///<</F<<///<//<F<F<BBFFF/FFFFFFBF/FBFF<FFFFBBBB/FBB<7FF/<BBBFB<BFFFFF/7<<7/BBFF/7B7/7/B7B//7B/BB	MD:Z:6T68T9G30	PG:Z:MarkDuplicates	RG:Z:22606_3#76	NH:i:1	HI:i:1	NM:i:3	AS:i:108
HS34_22606:3:1316:9877:80870	355	chr1	564464	3	8S115M	=	564664	324	GAGTACATGGGAGTCCGAACTAGTCTCAGGCTTCAACATCGAATACGCCGCAGGCCCCTTCGCCCTATTCTTCATAGCCGAATACACAAACATTATTATAATAAACACCCTCACCACTACAAT	/BBBB<FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF<FFFFFFFFFFFFFFFFFFFFFFFFBFFFFFFFFFFFFFFFFFFFFFFF/BFFFFFFFFFFFFFFFBFFFFFFFF<F	MD:Z:115	PG:Z:MarkDuplicates	RG:Z:22606_3#76	NH:i:2	HI:i:2	NM:i:0	AS:i:237
HS34_22606:3:1213:5571:77912	163	chr1	564659	3	125M	=	564673	139	CCCTGTTCTTATGAATTCGAACAGCATACCCCCGATTCCGCTACGACCAACTCATACACCTCCTATGAAAAAACTTCCTACCACTCACCCTAGCATTACTTATATGATATGTCTCCATACCCATT	BBBBBFBFFFFFFFF/FFFFFFFFFFFFFFFFFFFFBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF<FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFBFFB	MD:Z:125	PG:Z:MarkDuplicates	RG:Z:22606_3#76	NH:i:2	HI:i:1	NM:i:0	AS:i:248
```

I think I'm on the most up-to-date dev branch. The output of pip freeze for velocyto version is: 

```
-e git+https://github.com/velocyto-team/velocyto.py.git@e7b2a82dc8d23f47d190f7c4d3f326892d5ba2b8#egg=velocyto
```

Thanks for making the software, and for releasing as a preprint for people to read about and try while still under development.

**Comment by gioelelm** [comment]
The problem is that you are using `run` instead of `run_smartseq2`.
`run` has a more tricky set of inputs you need to get right.

The usage is
```
Usage: velocyto run_smartseq2 [OPTIONS] BAMFILES... GTFFILE

  Runs the velocity analysis on SmartSeq2 data (independent bam file per cell)

  [BAMFILES, ...] a sequence of bam files to be analyzed (e.g. use a wild-card expansion)

  GTFFILE genome annotation file

Options:
  -o, --outputfolder PATH  Output folder, if it does not exist it will be created.
  -e, --sampleid PATH      The sample name that will be used as a the filename of the output.
  -m, --repmask PATH       .gtf file containing intervals to mask
  -d, --dump TEXT          For debugging purposes only: it will dump a molecular mapping report to hdf5. --dump N, saves a cell every N cells. If p is prepended a more complete (but huge) pickle report is printed (default: 0)
  -v, --verbose            Set the vebosity level: -v (only warinings) -vv (warinings and info) -vvv (warinings, info and debug)
  --help                   Show this message and exit.
```
Furthermore you can pass al the bamfiles at the same time using wildcard (glob) expansion.

I am sorry if this is not in the docs yet but it will come soon.

**Comment by danielseaton** [comment]
Great, thanks, that works! There were a couple of side issues, which I'll make other issues for (I guess that's the easiest way for you to keep track of issues?)

Something unusual still happens when I try to run it on one individual bam file. So this works:

```
velocyto run_smartseq2 -o ./velocyto star/22606_3#73/22606_3#73.2pass.Aligned.sortedByCoord.dedup.bam star/22606_3#72/22606_3#72.2pass.Aligned.sortedByCoord.dedup.bam $GTF
```

But this gives the same IOError as above, saying it should be run with "-U":

```
velocyto run_smartseq2 -o ./velocyto star/22606_3#73/22606_3#73.2pass.Aligned.sortedByCoord.dedup.bam $GTF
```

But it obviously makes most sense to run lots of bams at once, so this isn't really a problem. I was just trying one at a time to test it out quickly.

>I am sorry if this is not in the docs yet but it will come soon.

No worries, didn't expect complete docs while trying out the dev branch!

**Comment by gioelelm** [comment]
Problem fixed in the dev branch. It will be released together wiht 0.17.1
